src/main/java/controllers/TopicController.java

```diff
@@ -20,39 +18,22 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
 
-        SearchDTO searchDTO = new SearchDTO();
+        TopicService topicService = new TopicService();
+        TopicService dropdown = new TopicService();
 
         String tagId = req.getParameter("tagId");
```

Comment:
I would suggest send request to service and get all parameters in the service

Reply:
Will discuss about it